apprise/plugins/NotifyThreema.py

```diff
@@ -0,0 +1,371 @@
+        }
+
+        # Create a copy of the targets list
+        targets = list(self.targets)
+
+        while len(targets):
+            # Get our target to notify
+            key, target = targets.pop(0)
+
+            # Prepare a payload object
+            payload = _payload.copy()
+
+            # Set Target
+            payload[key] = target
+
+            # Some Debug Logging
+            self.logger.debug(
+                'Threema Gateway GET URL: {} (cert_verify={})'.format(
+                    self.notify_url, self.verify_certificate))
+            self.logger.debug('Threema Gateway Payload: {}' .format(payload))
+
+            # Always call throttle before any remote server i/o is made
+            self.throttle()
+
+            try:
+                r = requests.post(
+                    self.notify_url,
+                    params=payload,
+                    headers=headers,
+                    verify=self.verify_certificate,
+                    timeout=self.request_timeout,
+                )
+
+                if r.status_code != requests.codes.ok:
+                    # We had a problem
+                    status_str = \
+                        NotifyThreema.http_response_code_lookup(
+                            r.status_code)
+
+                    self.logger.warning(
+                        'Failed to send Threema Gateway notification to {}: '
+                        '{}{}error={}'.format(
+                            target,
+                            status_str,
+                            ', ' if status_str else '',
+                            r.status_code))
+
+                    self.logger.debug(
+                        'Response Details:\r\n{}'.format(r.content))
+
+                    # Mark our failure
+                    has_error = True
+                    continue
+
+                # We wee successful
+                self.logger.info(
+                    'Sent Threema Gateway notification to %s' % target)
+
+            except requests.RequestException as e:
+                self.logger.warning(
+                    'A Connection error occurred sending Threema Gateway:%s '
+                    'notification' % target
+                )
+                self.logger.debug('Socket Exception: %s' % str(e))
+
+                # Mark our failure
+                has_error = True
+                continue
+
+        return not has_error
+
+    def url(self, privacy=False, *args, **kwargs):
+        """
+        Returns the URL built dynamically based on specified arguments.
+        """
+
+        # Define any URL parameters
+        params = self.url_parameters(privacy=privacy, *args, **kwargs)
+
+        schemaStr =  \
+            '{schema}://{gatewayid}@{secret}/{targets}?{params}'
+        return schemaStr.format(
+            schema=self.secure_protocol,
+            gatewayid=self.pprint(
+                self.user, privacy, mode=PrivacyMode.Secret, safe=''),
```

Comment:
Apprise builds its configuration from a URL as you've seen/know. But it can also reconstruct the URL for 2 use cases:
- public display: hide credentials from anyone looking over your shoulder. This plays a big roll with the [Apprise API](https://github.com/caronc/apprise-api) which displays back to you the URLs loaded. The privacy switches are all enabled and the URL doesn't leak anything someone shouldn't see. Logs can show this view too.
- private: the URL is reconstructed and is useful for serialization (pickle) and other cases where you want to turn your object back into the URL it was fed. In this case, all of those privacy wrappers are ignored and the contents are passed as is.

Long story short, there are no worries here. But if you think removing the gateway id from the public masking does no harm, i can easily switch that.

Reply:
Thanks for the insight!

The Gateway ID is a public username that so happens to also be part of the API authentication, but there is no way to obscure it (anyone with knowledge of the API docs will know it is used as part of API auth). The "secret" parameter is the real credential. So we might as well handle the Gateway ID as insensitive information.